rmw_fastrtps_dynamic_cpp/include/rmw_fastrtps_dynamic_cpp/TypeSupport_impl.hpp

```diff
@@ -844,28 +845,33 @@ size_t TypeSupport<MembersType>::calculateMaxSerializedSize(
       }
     }
 
+    last_member_size = 0;
     switch (member->type_id_) {
       case ::rosidl_typesupport_introspection_cpp::ROS_TYPE_BOOL:
       case ::rosidl_typesupport_introspection_cpp::ROS_TYPE_BYTE:
       case ::rosidl_typesupport_introspection_cpp::ROS_TYPE_UINT8:
       case ::rosidl_typesupport_introspection_cpp::ROS_TYPE_CHAR:
       case ::rosidl_typesupport_introspection_cpp::ROS_TYPE_INT8:
+        last_member_size = sizeof(int8_t);
         current_alignment += array_size * sizeof(int8_t);
         break;
       case ::rosidl_typesupport_introspection_cpp::ROS_TYPE_INT16:
       case ::rosidl_typesupport_introspection_cpp::ROS_TYPE_UINT16:
+        last_member_size = sizeof(uint16_t);
         current_alignment += array_size * sizeof(uint16_t) +
           eprosima::fastcdr::Cdr::alignment(current_alignment, sizeof(uint16_t));
         break;
       case ::rosidl_typesupport_introspection_cpp::ROS_TYPE_FLOAT32:
       case ::rosidl_typesupport_introspection_cpp::ROS_TYPE_INT32:
       case ::rosidl_typesupport_introspection_cpp::ROS_TYPE_UINT32:
+        last_member_size = sizeof(uint32_t);
         current_alignment += array_size * sizeof(uint32_t) +
           eprosima::fastcdr::Cdr::alignment(current_alignment, sizeof(uint32_t));
         break;
       case ::rosidl_typesupport_introspection_cpp::ROS_TYPE_FLOAT64:
       case ::rosidl_typesupport_introspection_cpp::ROS_TYPE_INT64:
       case ::rosidl_typesupport_introspection_cpp::ROS_TYPE_UINT64:
+        last_member_size = sizeof(uint64_t);
```

Comment:
```suggestion
        last_member_size = array_size * sizeof(uint64_t);
```

Reply:
Fixed in [d153fae](https://github.com/ros2/rmw_fastrtps/pull/716/commits/d153fae302171d43e2d4f5e705f88fe4cd0aa603)